packages/wallets/src/solana/wallets/base.ts

```diff
@@ -0,0 +1,51 @@
+import { Ecosystem, GenericSignerWallet } from "../../core";
+import { SolanaSigner } from "../interfaces/signer";
+import bs58 from "bs58";
+import nacl from "tweetnacl";
+
+export abstract class AbstractSigner implements GenericSignerWallet {
```

Comment:
I feel like you're using Signer and Wallet interchangeably and its confusing.

let's use:
- Signer only for things that implement ethers/solana Signer directly and are returned by `wallet.getSigner()`
- Wallet is an object that has a `getSigner()` method, and does wallet things like getAddress(), signMessage() etc